### ISSUE TITLE
network: Uncaught TypeError in NetworkManager version detection

### ISSUE THREAD

**Issue by piiiggg** [cockpit-project/cockpit#11346]
Cockpit version: 164
OS: Ubuntu Server 18.04.2 LTS
Page: Network

No networking info and font error on Chrome console: http://prntscr.com/mvg2i7

**Comment by garrett** [comment]
Here's a local version of the screenshot:

![font-error-networking](https://user-images.githubusercontent.com/10246/54139103-1b0da780-4421-11e9-888b-0477a81eeed6.png)

---

I don't think the problem is font-related whatsoever. The CSP font problem would, at worst, cause the font displayed on the page to just be wrong — and I don't see a font-related issue in the screenshot. (That said: There probably is a CSP error somewhere with loading _some_ font.)

The big issue I do see is the Uncaught TypeError in the `network.js` file.

It seems to be caused by `pkg/networkmanager/interfaces.js` in the following code:

```js
  self.at_least_version = function at_least_version(version) {
    return compare_versions(self.get_manager().Version, version) >= 0;
  };
```

which calls `compare_versions`, that has this line:

```js
      return str.split(".").map(function (s) {
```

So I'm guessing the `self.get_manager().Version` might be failing due to an invalid string or it may have an unexpected version.

**Comment by garrett** [comment]
@piiiggg: Thanks for reporting the issues!

I've opened another issue about the font loading errors — but what you're showing here in your console is more serious, so I've reworded it and changed the focus of the issue.

**Comment by garrett** [comment]
@piiiggg: I've tried to reproduce the error here with Ubuntu 18.04.2 LTS.

After doing a fresh install and installing Cockpit, the "Networking" page doesn't work here too! But I get different errors in the console.

_However_, after doing `apt update && apt upgrade` as root and rebooting, Cockpit's Networking page works here.

Can you please make sure your system is upgraded and reboot and see if that fixes things?

**Comment by piiiggg** [comment]
> @piiiggg: I've tried to reproduce the error here with Ubuntu 18.04.2 LTS.
> 
> After doing a fresh install and installing Cockpit, the "Networking" page doesn't work here too! But I get different errors in the console.
> 
> _However_, after doing `apt update && apt upgrade` as root and rebooting, Cockpit's Networking page works here.
> 
> Can you please make sure your system is upgraded and reboot and see if that fixes things?

The networking working after I reported this problem about a few hours, but still when I clicked to add VLANs. It happened again

http://prntscr.com/mx60yw

**Comment by garrett** [comment]
@piiiggg: Thanks for the update. What's happening now is really weird. Perhaps @mvollmer might know what's going on.

Meanwhile: Do you have any browser extensions installed?

**Comment by piiiggg** [comment]
> @piiiggg: Thanks for the update. What's happening now is really weird. Perhaps @mvollmer might know what's going on.
> 
> Do you have any browser extensions installed?

I do have a lot of extensions installed on Chrome, I also just try again with incognito mode (no addon) and nothing change. I still got the error.

**Comment by martinpitt** [comment]
Fixed in PR #13051 in commit ee8545f644a8627c847b2c4aa74c66550dac3be5
